File: jeonghyeon/8week/덧칠하기.js
function solution(n, m, section) {
    let count = 0; // 칠하는 수
    let fill = 0; // 마지막 칠한 위치
    for(let i = 0; i < section.length; i++) {
        if(section[i] >= fill) { // 현재 위치가 이미 칠해진 마지막 위치보다 크거나 같은 경우
            fill = section[i] + m; // 현재 위치부터 m만큼 칠하고, 그 위치 마지막 위치로 저장
            count++;
        }
    }
    return count;
}

File: gang/week8/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 1; // 칠한 횟수 기본 값으로 1

        int prev = section[0];  // 이전 구역의 끝 위치

        for (int i = 1; i < section.length; i++) {
            int gap = section[i] - prev;    // 현재 시작 위치와 이전 구역의 끝 위치 사이 계산
            if (gap >= m) { // 롤러의 길이보다 크면
                prev = section[i];  // 현재 시작 위치를 이전 구역의 끝 위치로 지정
                answer++;   // 칠한 횟수 추가
            }
        }

        return answer;
    }
}
